src/constants.ts

```diff
@@ -0,0 +1,8 @@
+import { readFileSync } from 'fs';
+import { dirname, join } from 'path';
+import { puppeteerDirname } from './compat.js';
+
+export const rootDirname = dirname(dirname(dirname(puppeteerDirname)));
+export const packageVersion = JSON.parse(
```

Comment:
how does it work in the bundled versions when there is no package.json?

Reply:
Good catch, I guess hoisting this constant up would cause errors in such an environment.

Reply:
I think it was not an issue when we required json as bundlers could probably trace package.json. But I don't think they will handle fs.read. Is there an alternative way to get the version for both esm/cjs?

Reply:
As discussed offline, we are generating the file.